tasks/configure.js

```diff
@@ -97,76 +106,62 @@ function task (opts) {
 		});
 	}
 
-	let authorizedPostHeaders = {
-		'Accept': 'application/vnd.heroku+json; version=3',
-		'Content-Type': 'application/json'
-	};
-
-	return herokuAuthToken()
-		.then(function (key) {
-			authorizedPostHeaders.Authorization = 'Bearer ' + key;
-
-			return getServiceData(source, opts.registry).then(serviceData => Promise.all([
-				fetchFromVault(source, target, serviceData),
-				fetch('https://api.heroku.com/apps/' + target + '/config-vars', { headers: authorizedPostHeaders })
-					.then(fetchres.json)
-					.catch(function (err) {
-						if (err instanceof fetchres.BadServerResponseError && err.message === 404) {
-							throw new Error(source + ' app needs to be manually added to heroku before it, or any branches, can be deployed');
-						} else {
-							throw err;
-						}
-					})
-				])
-				.then(function (data) {
-					let desired = data[0];
-					let current = data[1];
-
-					desired['SYSTEM_CODE'] = serviceData.code;
-
-					desired['___WARNING___'] = 'Don\'t edit config vars manually. Use the Vault UI.';
-					let patch = {};
-
-					Object.keys(current).forEach(function (key) {
-						if (!key.startsWith('HEROKU_') && !FORBIDDEN_ATTACHMENT_VARIABLES.includes(key)) {
-							patch[key] = null;
-						}
-					});
-
-					Object.keys(desired).forEach(key => {
-						if (FORBIDDEN_ATTACHMENT_VARIABLES.includes(key)) {
-							throw new Error(`\nCannot set environment variable '${key}' as this variable name is used for an attachment variable by Heroku, `
-								+ 'if this is for an external service, please use a different environment variable name in your app\n');
-						} else {
-							patch[key] = desired[key];
-						}
-					});
-
-					Object.keys(overrides).forEach(function (key) {
-						patch[key] = overrides[key];
-					});
-
-					Object.keys(patch).forEach(function (key) {
-						if (patch[key] === null) {
-							console.log('Deleting config var: ' + key); // eslint-disable-line no-console
-						} else if (patch[key] !== current[key]) {
-							console.log('Setting config var: ' + key); // eslint-disable-line no-console
-						}
-					});
-
-					console.log('Setting environment keys', Object.keys(patch)); // eslint-disable-line no-console
-
-					return fetch('https://api.heroku.com/apps/' + target + '/config-vars', {
-						headers: authorizedPostHeaders,
-						method: 'patch',
-						body: JSON.stringify(patch)
-					});
-				})
-				.then(response => {
-					if (response.status !== 200) return response.json().then(({id, message}) => Promise.reject(new Error(`Heroku Error - id: ${id}, message: ${message}`)));
-					console.log(target + ' config vars are set'); // eslint-disable-line no-console
-				}));
-			});
+	console.log('Retrieving pipeline details from Heroku...'); // eslint-disable-line no-console
+
+	// TODO: Investigate what herokuAuthToken is doing
+	const [ authToken, pipelineId ] = await Promise.all([
```

Comment:
Reminder to check if we can use the `herokuAuthToken` function more efficiently. Is it currently making redundant API calls using the heroku CLI?

Reply:
herokuAuthToken is getting the logged in user and then their token using the CLI. It is currently being called twice in configure.js as lib/pipelines calls it aswell, so will look into changing this.